core/src/main/java/org/bitcoinj/coinjoin/CoinJoinClientQueueManager.java

```diff
@@ -79,7 +86,10 @@ public void processDSQueue(Peer from, CoinJoinQueue dsq, boolean enable_bip61) {
                 log.info("coinjoin: DSQUEUE -- lastDsq: {}  dsqThreshold: {}  dsqCount: {}", nLastDsq, nDsqThreshold, context.masternodeMetaDataManager.getDsqCount());
                 // don't allow a few nodes to dominate the queuing process
                 if (nLastDsq != 0 && nDsqThreshold > context.masternodeMetaDataManager.getDsqCount()) {
-                    log.info("coinjoin: DSQUEUE -- Masternode {} is sending too many dsq messages", dmn.getProTxHash());
+                    if (!spammingMasternodes.containsKey(dsq.getProTxHash())) {
+                        spammingMasternodes.put(dsq.getProTxHash(), Utils.currentTimeMillis());
+                        log.info("coinjoin: DSQUEUE -- Masternode {} is sending too many dsq messages", dmn.getProTxHash());
+                    }
```

Comment:
The goal here is to reduce the logging of this message.  Allow it to be printed once.